knimin/lib/data_access.py

```diff
@@ -1685,6 +1685,34 @@ def add_external_survey(self, survey, description, url):
                  RETURNING external_survey_id"""
         return self._con.execute_fetchone(sql, [survey, description, url])[0]
 
+    def list_ag_surveys(self, selected=None):
+        """Returns the list of american gut survey names
+
+        Parameters
+        ----------
+        selected : list of int
+            To keep track of which surveys have been selected by the user via
+            the rendered website, the resulting list of tupels third element
+            holds this information as a boolean value.
+            If None is supplied all surveys are marked as being selected, if
+            a list of IDs (in form of ints) is supplied, only those surveys are
+            marked with True that exist in the DB and have a matching ID to one
+            of the given IDs.
+
+        Returns
+        -------
+        list of (int, str, bool)
```

Comment:
This function is used to render all survey names on the metadata pulldown website. The user (who is browsing labadmin) has the option to tick a survey to include or exclude it for the actual pulldown.
Thus, this function is a hybrid of DB query and interface rendering, but since it is so simple I thought it would be OK.